pulp_2to3_migration/app/plugin/rpm/migrator.py

```diff
@@ -126,6 +158,108 @@ def pipeline_stages(self):
         return pipeline
 
 
+class RpmContentMigrationFirstStage(ContentMigrationFirstStage):
+    """
+    The first stage of a Rpm content migration pipeline.
+    """
+
+    async def migrate_to_pulp3(self, batch, pb=None):
+        """
+        A default implementation of DeclarativeContent creation for migrating content to Pulp 3.
+
+        Plugin writers might want to override this method if it doesn't satisfy their needs as is.
+
+        In this implementation there is an assumption that each content has one artifact.
+
+        Args:
+            batch: A batch of Pulp2Content objects to migrate to Pulp 3
+        """
+        @functools.lru_cache(maxsize=20)
+        def get_remote_by_importer_id(importer_id):
+            """
+            Args:
+                importer_id(str): Id of an importer in Pulp 2
+
+            Returns:
+                remote(pulpcore.app.models.Remote): A corresponding remote in Pulp 3
+
+            """
+            try:
+                pulp2importer = Pulp2Importer.objects.get(pulp2_object_id=importer_id)
+            except ObjectDoesNotExist:
+                return
+            return pulp2importer.pulp3_remote
+
+        for pulp_2to3_detail_content in batch:
+            pulp2content = pulp_2to3_detail_content.pulp2content
+
+            if getattr(pulp2content, 'downloaded'):
```

Comment:
Why ```getattr```, shouldn't all ```pulp2content``` have the ```downloaded``` field?

Reply:
not all of them devire from FileContentUnit, some of them a fileless https://github.com/pulp/pulp-2to3-migration/blob/master/pulp_2to3_migration/pulp2/base.py#L11 VS https://github.com/pulp/pulp-2to3-migration/blob/master/pulp_2to3_migration/pulp2/base.py#L36